File: GXpert/GXpert.Web/Modules/Exams/Assignment/AssignmentForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Exams.Forms;

[FormScript("Exams.Assignment")]
[BasedOnRow(typeof(AssignmentRow), CheckNames = true)]
public class AssignmentForm
{
    public string Title { get; set; }
    [TextAreaEditor(Rows =3)]
    public string Description { get; set; }
    [FileUploadEditor]
    public string File { get; set; }
    public int ClassId { get; set; }
    public int SubjectId { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Exams/AssignmentForm.ts
import { StringEditor, TextAreaEditor, ImageUploadEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface AssignmentForm {
    Title: StringEditor;
    Description: TextAreaEditor;
    File: ImageUploadEditor;
    ClassId: LookupEditor;
    SubjectId: LookupEditor;
}

export class AssignmentForm extends PrefixedContext {
    static readonly formKey = 'Exams.Assignment';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AssignmentForm.init)  {
            AssignmentForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = ImageUploadEditor;
            var w3 = LookupEditor;

            initFormType(AssignmentForm, [
                'Title', w0,
                'Description', w1,
                'File', w2,
                'ClassId', w3,
                'SubjectId', w3
            ]);
        }
    }
}